src/components/modals/MigrateAccounts/index.js

```diff
@@ -0,0 +1,201 @@
+  state = INITIAL_STATE
+  STEPS = createSteps()
+
+  handleStepChange = (step: Step) => this.setState({ stepId: step.id })
+  handleSetAppOpened = (isAppOpened: boolean) => this.setState({ isAppOpened })
+  handleAddMigratedAccounts = (currency, account) => {
+    this.setState(prevState => ({
+      migratedAccounts: {
+        ...prevState.migratedAccounts,
+        [currency.id]: [...(prevState.migratedAccounts[currency.id] || []), account],
+      },
+    }))
+  }
+  handleSetScanStatus = (scanStatus: string, err: ?Error = null) => {
+    if (err) {
+      logger.critical(err)
+    }
+    this.setState({ scanStatus, err })
+  }
+
+  handleNextCurrency = () => {
+    const { migratableAccounts } = this.props
+    const { migratedAccounts } = this.state
+    const currencyId = Object.keys(migratableAccounts).find(
+      d => !Object.keys(migratedAccounts).includes(d),
+    )
+
+    const currencyAccounts = migratableAccounts[currencyId]
+    const currency = currencyAccounts && currencyAccounts[0].currency
+
+    this.setState({ currency })
+  }
+
+  handleFlushMigratedAccounts = async () => {
+    await this.setState({ migratedAccounts: {} }, this.handleNextCurrency)
+  }
+
+  handleCloseModal = () => this.props.closeModal(MODAL_MIGRATE_ACCOUNTS)
+
+  render() {
+    const {
+      device,
+      migratableAccounts,
+      totalMigratableAccounts,
+      accounts,
+      replaceAccounts,
+    } = this.props
+    const { stepId, isAppOpened, err, scanStatus, migratedAccounts, currency } = this.state
+
+    const stepperProps = {
+      replaceAccounts,
+      migratableAccounts,
+      migratedAccounts,
+      totalMigratableAccounts,
+      accounts,
+      device,
+      currency,
+      isAppOpened,
+      err,
+      scanStatus,
+      setAppOpened: this.handleSetAppOpened,
+      setScanStatus: this.handleSetScanStatus,
+      nextCurrency: this.handleNextCurrency,
+      addMigratedAccount: this.handleAddMigratedAccounts,
+      flushMigratedAccounts: this.handleFlushMigratedAccounts,
+      onCloseModal: this.handleCloseModal,
+    }
+
+    const errorSteps = err ? [2] : []
+    const disableDismiss = scanStatus === 'scanning'
+
+    return (
+      <Modal
+        centered
+        name={MODAL_MIGRATE_ACCOUNTS}
+        preventBackdropClick={disableDismiss}
+        onHide={disableDismiss ? undefined : () => this.setState({ ...INITIAL_STATE })}
+        render={({ onClose }) => (
+          <Stepper
+            hideBreadcrumb
+            initialStepId={stepId}
+            onStepChange={this.handleStepChange}
+            onClose={disableDismiss ? undefined : onClose}
+            steps={this.STEPS}
+            errorSteps={errorSteps}
+            {...stepperProps}
+          />
+        )}
+      />
+    )
+  }
+}
+
+const mapStateToProps = createStructuredSelector({
+  device: getCurrentDevice,
+  accounts: accountsSelector,
+  totalMigratableAccounts: state => accountsSelector(state).filter(canBeMigrated).length,
```

Comment:
I think this is not necessary because it's already part of `migratableAccounts` . i would prefer we don't do twice the same thing . either we could just inject `accountsSelector(state).filter(canBeMigrated)` as one prop, or we could just infer that total accounts by doing a sum of all groups. if it's just about checking if there are some remaining, would it be the same as checking `Object.keys(migratableAccounts)` length?

Reply:
According to design we need the "There are x accounts left to migrate" after a device loop. I'd need to iterate all keys in `migratableAccounts`, and sum them. I had it like that, considered this approach cleaner since we already have the `accountsSelector` there